src/lib.rs

```diff
@@ -670,7 +737,29 @@ impl Library {
                         self.include_paths.push(PathBuf::from(inc));
                     }
                 }
-                _ => (),
+                _ => {
+                    let path = std::path::Path::new(part);
+                    if path.is_file() {
+                        // Cargo doesn't have a means to directly specify a file path to link,
+                        // so split up the path into the parent directory and library name.
+                        // TODO: pass file path directly when link-arg library type is stabilized
+                        // https://github.com/rust-lang/rust/issues/99427
+                        if let Some(dir) = path.parent() {
+                            let link_search = format!("rustc-link-search={}", dir.display());
+                            config.print_metadata(&link_search);
+                        }
+                        if let (Some(file_name), Ok(target)) = (path.file_name(), &target) {
+                            if let Some(lib_basename) = Self::extract_lib_from_filename(
```

Comment:
If this fails then a warning should probably be printed so that files are not silently skipped. Or should it maybe even be an error as linking will likely fail with unresolved symbols now?

Reply:
The build script can't know if omitting this will fail to link, but printing a warning is reasonable I think.